app/controllers/application_controller.rb

```diff
@@ -57,7 +57,7 @@ def restrict_iframes
   def enable_embedded_shopfront
     whitelist = Spree::Config[:embedded_shopfronts_whitelist]
     return unless Spree::Config[:enable_embedded_shopfronts] && whitelist.present?
-    return if request.referer && URI(request.referer).scheme != 'https' && !Rails.env.test?
+    return if request.referer && URI(request.referer).scheme != 'https' && !Rails.env.test? && !Rails.env.development?
```

Comment:
shouldn't we use `force_ssl` in production config? Isn't the whole site using SSL? This way we don't need to manually check any env. That would be set in its config file.

Reply:
This is here because of issues with non-SSL sites embedding OFN pages, to make sure they are using SSL.

Reply:
what kind of issues?

Reply:
This line checks the referer (if present) to see if the embedding website uses https. Otherwise a sneaky man-in-the-middle wifi router could inject some Javascript code into the embedding http website. That code could then access contents in the iframe and mess around with payment methods or create orders for the attacker.

My change relaxes the condition so that I can test in development mode without https.

Reply:
My point is that why do we need to check only here whether the request is HTTPS? Isn't the whole site using SSL? Then we don't need to make this controller aware of SSL but just the app config (as I guess it is now).

Then, the for test or dev env would be useless as that is configured in each env config file for the whole app.

Feel free to dismiss my comment, I might be missing something.

Reply:
> Isn't the whole site using SSL?

All staging and production OFN instances are using SSL/TLS, yes. But the `referer` here is the URL of the embedding website, a third party we don't control, not OFN.  But it is important that that other website uses SSL/TLS as well.